.github/workflows/docker-build-and-publish.yaml

```diff
@@ -67,11 +68,14 @@ on:
         description: "GitHub token."
         required: false
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}-docker
+  cancel-in-progress: true
```

Comment:
Why did you change the `cancel-in-progress` everywhere? The idea is to only cancel the workflow when we are not on the default branch, so that we still let every pipeline run for every merge to main (even if they are running at the same time)